migrations/custom/openrecords_v1.py

```diff
@@ -1,107 +1,1136 @@
+        title = record.filename
+    else:
+        title = record.description
+
+    CUR_V2.execute(query, (
+        response_id,
+        title,
+        record.filename,
+        mime_type,
+        hash_,
+        size
+    ))
+
+
+@transfer('Links', "SELECT * FROM record WHERE url IS NOT NULL and url != '1'")
+def transfer_links(record):
+    response_id = _create_response(record, response_type.LINK,
+                                   _get_record_release_date(record))
+
+    query = ("INSERT INTO links ("
+             "id, "
+             "title, "
+             "url) "
+             "VALUES (%s, %s, %s)")
+
+    CUR_V2.execute(query, (
+        response_id,
+        record.description,
+        record.url
+    ))
+
+
+@transfer('Instructions', "SELECT * FROM record WHERE access IS NOT NULL")
+def transfer_instructions(record):
+    """
+    For records with non-null description fields, the description and access
+    content are joined and stored in instructions.content.
+    """
+    response_id = _create_response(record, response_type.INSTRUCTIONS,
+                                   _get_record_release_date(record))
+
+    query = ("INSERT INTO instructions ("
+             "id, "
+             "content) "
+             "VALUES (%s, %s)")
+
+    if record.description is None or record.description.strip() == '':
+        content = record.access
+    else:
+        content = ':\n\n'.join((record.description, record.access))
+
+    CUR_V2.execute(query, (
+        response_id,
+        content
+    ))
+
+
+# TODO: Handle emails where recipient does not have an email address?
+@transfer('Emails', "SELECT * "
+                    "FROM email_notification "
+                    "WHERE recipient NOT IN (SELECT id "
+                    "                        FROM public.user "
+                    "                        WHERE email is NULL)")
+def transfer_emails(email):
+    response_id = _create_response(email, response_type.EMAIL,
+                                   release_date=None,
+                                   privacy=response_privacy.PRIVATE,
+                                   date_created=email.time_sent)
+
+    query = ("INSERT INTO emails ("
+             "id, "
+             '"to", '
+             "subject, "
+             "body) "
+             "VALUES (%s, %s, %s, %s)")
+
+    CUR_V1.execute("SELECT email FROM public.user WHERE id = %s"
+                   % email.recipient)
+    to = CUR_V1.fetchone().email
+
+    CUR_V2.execute(query, (
+        response_id,
+        to,
+        email.subject,
+        email.email_content.get('email_text')
+    ))
+
+
+@transfer('Users', "SELECT * FROM public.user "
+                   "WHERE alias NOT IN (SELECT name FROM department) "
+                   "      AND alias IS NOT NULL "
+                   "      OR (first_name IS NOT NULL "
+                   "          AND last_name IS NOT NULL)")
+def transfer_users(user_ids_to_guids, user):
```

Comment:
It looks like we are going to have to hardcode certain users to ignore (remember to return -1).